Rspec test

ioilmio opened the pull request:
We added test for Board and player class methods including:
---------------------
 For Player class: 
-unique?
-player_array
-player_totem
-win?
-tie?
-----------------
For Board class:
-valid?
-who_player

ioilmio committed:
Merge pull request #5 from guirecordon/development

All milestone reviewed and approved merging to master for production

guirecordon committed:
Add board_spec.rb file and write tests

ioilmio committed:
Add test for player class: unique?, player_array, player_totem, win?, tie?

suretrust commented on spec/board_spec.rb:
```diff
@@ -0,0 +1,26 @@
+require './lib/board'
+require './lib/player'
```
This should be `require_relative`. `require` should be used for external files, like gems, while `require_relative` should be used for referring to files within your directory.

ioilmio replied:
fixed

suretrust commented on spec/player_spec.rb:
```diff
@@ -0,0 +1,44 @@
+require './lib/board'
+require './lib/player'
```
This should be require_relative. require should be used for external files, like gems, while require_relative should be used for referring to files within your directory.

ioilmio replied:
fixed

suretrust reviewed (changes requested):
Hi team, 👋 

### Great job thus far! 🙌 However, some vitals are missing. 👇 

- [x] Add linter to your repository.

- [x] Let's update your readme to look more professional. Don't forget to add how to run the test.

**Kindly submit for another code review once these issues have been resolved.** _Happy coding!_ 🎉

ioilmio committed:
readme updated  and require_relative fix

ioilmio commented:
The readme get updated and the linter should be in place

guirecordon committed:
Run rubocop on all files

guirecordon committed:
Merge branch 'development' of github.com:guirecordon/tictactoe into rspec_test

guirecordon committed:
delete empty file

guirecordon committed:
Update README.md file

guirecordon committed:
Add run on repl.it badge to README

This pull request adds a `Run on Repl.it` badge to the `README`. This will allow users to easily run this repository in their browser, without having to set up an environment. You can learn more about Repl.it [here](https://repl.it).

guirecordon committed:
Merge pull request #7 from guirecordon/replit-readme

Replit readme

ioilmio committed:
removed require_relative statement for missing file game_logic

ioilmio committed:
merged rspec changes

ioilmio committed:
gitignore replit

ioilmio committed:
renamed main.rb to main

ioilmio committed:
updated dependecies

ioilmio committed:
repl.it link

ioilmio committed:
replit link

guirecordon committed:
Add Gemfile to rubocop yml exclude

mosesogwo reviewed (changes requested):
**Hi @ioilmio, @guirecordon, good work making the requested changes:+1:, but your test is now broken**

:point_right:  When you use require_relative, you should specify a relative part to the resource.
In your case, the path to your resource should start with two dots, referring to the parent folder before the  lib folder. For example `require_relative './lib/board'` should be `require_relative '../lib/board'`.

:point_right: **[Optional]** Also, your tests should cover both positive and negative scenarios. Sorry, this wasn't pointed out before now. For example, your tie and win methods should be tested for when they are expected to return false, all win conditions should be tested. 

_**Kindly make this change and verify that your tests are passing before submitting another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

guirecordon committed:
Change relative path

guirecordon committed:
Add negative test cases to board_spec and player_spec

guirecordon committed:
Fix linter errors one last time

guirecordon commented:
Thanks for this review, Moses. We fixed the relative paths and added negative test cases.

Redvanisation reviewed (approved):
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀
